exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -93,13 +93,15 @@ public void ensureAtLeastOneField(ComplexWriter writer) {
     if (!atLeastOneWrite) {
       // if we had no columns, create one empty one so we can return some data for count purposes.
       SchemaPath sp = columns.get(0);
-      PathSegment root = sp.getRootSegment();
+      PathSegment fieldPath = sp.getRootSegment();
       BaseWriter.MapWriter fieldWriter = writer.rootAsMap();
-      while (root.getChild() != null && !root.getChild().isArray()) {
-        fieldWriter = fieldWriter.map(root.getNameSegment().getPath());
-        root = root.getChild();
+      while (fieldPath.getChild() != null && ! fieldPath.getChild().isArray()) {
+        fieldWriter = fieldWriter.map(fieldPath.getNameSegment().getPath());
+        fieldPath = fieldPath.getChild();
+      }
+      if (fieldWriter.isEmptyMap()) {
```

Comment:
Let me restate:

We're using ensureAtLeastOneField to make sure we have at least one vector has been created. If we check this condition and it is false, then we do nothing. My question is: can we remove the ensureAtLeastOneField property and replace the condition above to be fieldWriter.isEmptyMap() instead?

Basically change:

if (!atLeastOneWrite) {

to

if (fieldWriter.isEmptyMap()) {